src/collaborative/ot/srt_specific.ts

```diff
@@ -101,3 +105,92 @@ function addPivotCommandAdaptRange<Cmd extends AddPivotCommand | UpdatePivotComm
 }
 specificRangeTransformRegistry.add("ADD_PIVOT", addPivotCommandAdaptRange);
 specificRangeTransformRegistry.add("UPDATE_PIVOT", addPivotCommandAdaptRange);
+
+function adaptChartRange<Cmd extends CreateChartCommand | UpdateChartCommand>(
+  cmd: Cmd,
+  applyChange: RangeAdapter
+): Cmd {
+  cmd = deepCopy(cmd);
+  const definition = cmd.definition;
+  const type = definition.type;
+  switch (type) {
+    case "bar":
+    case "line":
+    case "combo":
+    case "funnel":
+    case "pie":
+    case "waterfall":
+    case "pyramid":
+    case "radar":
+    case "sunburst":
+    case "treemap":
+    case "geo":
```

Comment:
I predict a 94.87% of chance that we will forget about this the next time we add a new chart type. Can't we do some `if "dataset" in definition` ?